### PR TITLE
MAC_Hints.rst: Instruction to install pyenv added

### PR DESCRIPTION
**Pull request by rhemon**
While doing fix for https://github.com/coala/coala/issues/3461 
I added Instruction to install pyenv and link to the issue @jayvdb mentioned

### CODE REVIEW

File: Help/MAC_Hints.rst [hunk 1]
```diff
@@ -13,7 +13,15 @@ Coverage Installation Hints for macOS Users:
     $ brew install python3
     $ python3 --version   # To check the installed version
 
-3. Create Virtual Environments with pyvenv
+3. Install pyenv with homebrew
```
Comment (Nosferatul) [inline Help/MAC_Hints.rst]: maybe we could add a note that explain why we use this?

Comment (Mixih) [inline Help/MAC_Hints.rst]: hmm, it should be explained in the main setup page... maybe a note @Nosferatul if you feel it is necessary?

**Comment by supergr35** [comment]
Please use "Closes" instead of "Fixes" in your commit message. Here's a link to explain why:
http://api.coala.io/en/latest/Developers/Writing_Good_Commits.html#issue-reference

**Comment by Adrianzatreanu** [comment]
Hey, could you please rebase and follow the comments advice? :) message us on coala.io/chat if you need any kind of help :D

**Comment by Warewolf96** [comment]
Hey, please follow this http://api.coala.io/en/latest/Developers/Writing_Good_Commits.html
Ask for help.on gitter channel if u want...

**Comment by jayvdb** [comment]
ack f4ee6c4

**Comment by jayvdb** [comment]
needs rebase

**Comment by SanketDG** [comment]
ack 783f1d6

**Comment by SanketDG** [comment]
unack 783f1d6

File: Help/MAC_Hints.rst [hunk 2]
```diff
@@ -13,7 +13,15 @@ Coverage Installation Hints for macOS Users:
...
+------------------------------
+
+::
+
+    $ brew update
+    $ brew install pyenv
+
+4. Create Virtual Environments with pyvenv
```
Comment (SanketDG) [inline Help/MAC_Hints.rst]: I am not sure I understand the change here.

pyenv is being told to install, but then `pyvenv` is being used? Those are two different things afaik.

Comment (rhemon) [inline Help/MAC_Hints.rst]: the module is called pyenv, but to call it they use pyvenv. https://github.com/yyuu/pyenv

Comment (SanketDG) [inline Help/MAC_Hints.rst]: No they do not? `pyvenv` is a deprecated script wrapper for `python -m venv` IIRC. Nothing to do with pyenv.

File: Help/MAC_Hints.rst [hunk 3]
```diff
@@ -28,7 +36,10 @@ Coverage Installation Hints for macOS Users:
     # You can start Python3 by typing:
     $ python
 
-4. Virtualenvwrapper with Python 3:
+Presently, tests for version-file-read fail, if interested you can
```
Comment (jayvdb) [inline Help/MAC_Hints.rst]: This should be in the `pyenv` section

File: Help/MAC_Hints.rst (hunk 1 above)
Comment (jayvdb) [inline Help/MAC_Hints.rst]: This should be instead of the section "2. Install Python3."

See https://gist.github.com/Bouke/11261620#gistcomment-1573091

**Comment by rhemon** [comment]
so now I have separated into a pyenv section but i have recommended to still setup a virtualenv for development purpose. is that alright? 
should I add a virtualenv section here? or I am think of giving instructions of virtualenv in Development_Setup so I could link it to there?

File: Help/MAC_Hints.rst [hunk 4]
```diff
@@ -1,25 +1,74 @@
...
 
+pyenv
+-----
+
+Here we use pyenv, which is simply a python installer that allows
+you to install multiple python versions and switch through them.
+In case your macOS has an older version of python that coala doesn't
+support we recommend using pyenv to install and shift to a newer
+supported version. After installation, you may also want to set up
+a virtual environment using :code:`venv` or :code:`virtualenv`.
```
Comment (Udayan12167) [inline Help/MAC_Hints.rst]: Probably add jayvdb's comment that pyenv is deprecated and it's recommended to use venv.

Comment (SanketDG) [inline Help/MAC_Hints.rst]: `pyenv` is not deprecated, its a third party tool to manage different versions of python.

I think given that `venv` just works, there is no point creating confusion and complexity by asking to install `pyenv`. Plus, the whole point of `pyenv` is to manage different versions of python and not shift to a newer supported version.

**Comment by rhemon** [comment]
sorry, presently bit busy with my studies, will do the changes as soon as possible

File: Help/MAC_Hints.rst (hunk 4 above)
Comment (rhemon) [inline Help/MAC_Hints.rst]: I think by default macOS has Python 2.7 installed (at least that was the case for me) so by saying "shift to newer supported version" I tried to mean that using pyenv they install python > 3.3 that is a newer and a version with which coala works. (and at the beginning I mentioned that pyenv is for installing different python version and switch through them)

now this may be helpful for people who have an older python version in their os, however there are other ways as well that users can do easily (like using homebrew)

so do I remove the pyenv part? @SanketDG @jayvdb

Comment (SanketDG) [inline Help/MAC_Hints.rst]: `pyenv` is a tool that solves a problem that originated because of the complexity of two python versions of the same machine. It's workflow is entirely different from conventional python tools.

I think just using homebrew to install/update python is fine, and does't create any additional complexity.

File: Help/MAC_Hints.rst [hunk 5]
```diff
@@ -1,25 +1,36 @@
...
----------------------------------------------------
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 2. Install Python3.
--------------------
+~~~~~~~~~~~~~~~~~~~
+
+For coala you will need to use python3, so you may
+simply use homebrew to install python3, or you could also
+refer to pyenv section to install python3 while you can also
```
Comment (hemangsk) [inline Help/MAC_Hints.rst]: "refer to pyenv section", this can be a hyperlink to the pyenv section below

File: Help/MAC_Hints.rst [hunk 6]
```diff
@@ -1,25 +1,36 @@
 Coverage Installation Hints for macOS Users:
 ============================================
 
+venv
+----
+
+Here we will be using venv, which is part of python's standard
```
Comment (Asnelchristian) [inline Help/MAC_Hints.rst]: "``venv``" should be in double back sticks

Comment (Asnelchristian) [inline Help/MAC_Hints.rst]: ``

File: Help/MAC_Hints.rst [hunk 7]
```diff
@@ -29,7 +40,7 @@ Coverage Installation Hints for macOS Users:
     $ python
 
 4. Virtualenvwrapper with Python 3:
------------------------------------
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (Asnelchristian) [inline Help/MAC_Hints.rst]: this should be in a different commit

File: Help/MAC_Hints.rst [hunk 8]
```diff
@@ -54,9 +65,51 @@ Coverage Installation Hints for macOS Users:
     $ mkvirtualenv --python=python3_path myenv
 
 Finally!
---------
+~~~~~~~~
```
Comment (Asnelchristian) [inline Help/MAC_Hints.rst]: same here

File: Help/MAC_Hints.rst [hunk 9]
```diff
@@ -1,25 +1,36 @@
...
+
 1. Make sure you have installed Xcode and Homebrew.
----------------------------------------------------
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 2. Install Python3.
--------------------
+~~~~~~~~~~~~~~~~~~~
+
+For coala you will need to use python3, so you may
```
Comment (sils) [inline Help/MAC_Hints.rst]: in texts please use Python 3 consistently, uppercase P, space before the 3. Appears multiple times wrongly across your commit

File: Help/MAC_Hints.rst (hunk 5 above)
Comment (sils) [inline Help/MAC_Hints.rst]: refer to *the* pyenv section?

**Comment by Asnelchristian** [comment]
Please would you also address @hemangsk issue as well... :sweat_smile: ?

File: Help/MAC_Hints.rst [hunk 10]
```diff
@@ -54,9 +65,51 @@ Coverage Installation Hints for macOS Users:
...
+To use a particular version you simply have to do this :
+
+::
+
+    $ pyenv local 3.5.0
+
+Presently, while using pyenv, tests for version-file-read fail,
+if interested you can
+have a look `here <https://github.com/yyuu/pyenv/issues/623>`__
+
```
Comment (sils) [inline Help/MAC_Hints.rst]: you have *two* newlines before eof, you'll want only one ;)

**Comment by sils** [comment]
unack https://github.com/coala/documentation/pull/381/commits/dd4369bd5e1a487c801c4cf08a0aebeadedd3a6b

**Comment by rhemon** [comment]
sorry for being late :pray:

File: Help/MAC_Hints.rst [hunk 11]
```diff
@@ -60,3 +71,44 @@ Finally!
...
+    $ pyenv install 3.2.6
+    $ pyenv install 2.7.10
+    $ pyenv versions # lists the versions you have
+
+To use a particular version you simply have to do this :
+
+::
+
+    $ pyenv local 3.5.0
+
```
Comment (SanketDG) [inline Help/MAC_Hints.rst]: probably add the list of commands link here 
https://github.com/pyenv/pyenv/blob/master/COMMANDS.md

**Comment by sils** [comment]
ack https://github.com/coala/documentation/pull/381/commits/a13e9abca4d588bd5fe65667f2ab8777883d9219 https://github.com/coala/documentation/pull/381/commits/2097bc27bc91f32b04b00bdb7ad2ae47c4d2dcf2

**Comment by sils** [comment]
@rultor merge
